lib/filter/filter-call.c

```diff
@@ -50,9 +50,10 @@ filter_call_eval(FilterExprNode *s, LogMessage **msgs, gint num_msg, LogTemplate
   else
     stats_counter_inc(self->super.not_matched);
 
+  msg_set_context(msgs[num_msg - 1]);
   msg_trace("filter() evaluation started",
-            evt_tag_str("called-rule", self->rule),
-            evt_tag_msg_reference(msgs[num_msg - 1]));
+            evt_tag_str("called-rule", self->rule));
+  msg_set_context(NULL);
```

Comment:
This will remove the reference to the current message, is that really a good idea? the next filter would stop printing the rcptid this way.

I don't think either msg_set_context( ) calls are needed here.

Reply:
I don't think we need to set the context this many times. Every case, the current rcptid will be the same as there's only one message being handled at any given time.

Since we are not printing the pointer anymore, I don't think any of the msg_set_context() calls are needed in this patch. Let's discuss.

Reply:
Attila and I discussed this IRL, here is the summarized version:

1. `msg_set_context()` calls should be placed at places where new `LogMessage` instances are encountered. It's constructor seemed a viable option at first, however that would introduce a layering violation so we opted for moving it elsewhere.
2. During our internal queuing where `pop()` calls happen is the moment in time where logs are encountered before being sent to destinations. This combined with setting context at insertions (reading from source) seemed like the best place for them.
3. `LogScheduler` has it's very own (and unique) way of fetching logs, that's why it required a separate patch (easy to miss it) .
4. We could not move it inside `log_source_post` as there are already logs printed before the call actually happens.